ctl/vr/inductive_complex.h

```diff
@@ -0,0 +1,142 @@
+        if (*ai < v) { lower_neighbors.push_back( *ai); }
+    }
+} 
+
+// We add the given graph to the complex. Also, we update the cells vectors to
+// keep track of the dimensions of the cells.
+template<typename Graph, typename Complex, typename Vector>
+void add_graph_to_complex (const Graph& g,
+			   Complex& complex,
+			   Vector& k_cells,
+			   Vector& k_plus_one_cells) {
+    typedef typename Complex::Cell Simplex;
+    typedef typename boost::graph_traits< Graph>::vertex_iterator vertex_iterator;
+    typedef typename boost::graph_traits< Graph>::adjacency_iterator Iterator;
+    vertex_iterator vi, vlast;
+    for ( std::tie( vi, vlast) = boost::vertices( g); vi != vlast; ++vi) {
+        // adding {*vi} to both k_cells and complex
+        Simplex tau(1, *vi);
+        k_cells.push_back(tau);
+        complex.insert_open_cell(tau);
+        Iterator ai, ai_end;
+        for ( boost::tie(ai, ai_end) = boost::adjacent_vertices(*vi, g);
+              ai != ai_end; ++ai) {
+                // adding {*vi, *ai} to both k_plus_one_cells and complex
+                Simplex sigma( tau);
+                sigma.insert( *ai);
+                k_plus_one_cells.push_back(sigma);
+                complex.insert_open_cell(sigma);
+        }
+    }
+}
+
+// We loop through all nodes, selecting each one as the starting point for
+// constructing our simplicial complex.
+template<typename Graph, typename Complex> 
+void inductive_vr (const Graph& g, Complex& complex, std::size_t dimension) { 
+    typedef typename Complex::Cell Simplex;
+    typedef typename Simplex::value_type Vertex;
+    typedef typename Simplex::const_iterator simplex_iterator;
+    std::vector<Simplex> k_cells;
+    std::vector<Simplex> k_plus_one_cells;
+    // adding the graph to the complex
+    add_graph_to_complex (g, complex, k_cells, k_plus_one_cells);
+    // inductive algorithm
+    for ( int k = 1; k < dimension; k++) {
+        k_cells.swap(k_plus_one_cells);
+	k_plus_one_cells.clear();
+	for ( int i = 0; i < k_cells.size(); i++) {
+	    Simplex tau(k_cells[i]);
+	    // getting the intersection of all lower neighbors
+	    std::vector< Vertex> final_neighbors;
+            std::vector< Vertex> lower_neighbors;
+	    get_lower_neighbors(g, *(tau.begin()), final_neighbors);
+	    for(simplex_iterator vi = tau.begin(); vi != tau.end(); ++vi) {
+		lower_neighbors.clear();
+		get_lower_neighbors(g, *vi, lower_neighbors);
+		std::set< Vertex> finalN(final_neighbors.begin(), final_neighbors.end());
+		std::set< Vertex> lowerN(lower_neighbors.begin(), lower_neighbors.end());
+		final_neighbors.clear();
+		set_intersection(lowerN.begin(),
+				 lowerN.end(),
+				 finalN.begin(),
+				 finalN.end(),
+				 back_inserter(final_neighbors));
+	    }
+	    // constructing new simplices and adding them to the complex
+    	    for( int j = 0; j < final_neighbors.size(); ++j) {
+		Simplex sigma( tau);
+		sigma.insert( final_neighbors[j]); 
+		k_plus_one_cells.push_back(sigma);
```

Comment:
I would try make use of emplace_back here and I would avoid the explicit loop.

e.g. you have  simplices tau and "final_neighbors" and you wish to take there union. 
You essentially want this:
k_plus_one_cells.emplace_back( union.begin(), union.end());

but: you just have the two simplices.

There are two ways to do this, you could manually write a concatenation iterator which takes the two begin(), end() pairs and turns them into one begin()/end() pair. 

e.g: http://stackoverflow.com/questions/981186/chain-iterator-for-c

OR

You could try something like const auto & union =  boost::range::join( tau, final_neighbors);

Finally, after you create the object in the k_plus_one_cells structure, then insert it into the complex...